src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -566,6 +570,61 @@ private bool DelegateNeedsReturn(MethodSymbol invokeMethod)
             return true;
         }
 
+        private BoundLambda ReallyBindNaturalType(TypeSymbol expressionType)
```

Comment:
nit: consider calling the parameter "delegateType" to match caller and other similar method

Reply:
In this case, the type is not a (strongly-typed) delegate type though. It is either `System.Delegate` or `System.Linq.Expressions.Expression`. I realize the naming is inconsistent since callers refer to this as "delegateType" but I was reluctant to change the callers. The current names were a compromise although not ideal.